cloud-logging entry iteration example broken, fails if more than 1 page.

Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

Please run down the following list and make sure you've tried the usual "quick fixes":

  - Search the issues already opened: https://github.com/googleapis/java-logging/issues
  - Check for answers on StackOverflow: http://stackoverflow.com/questions/tagged/google-cloud-platform

If you are still having issues, please include as much information as possible:

#### Environment details

1. Specify the API at the beginning of the title. For example, "BigQuery: ...").
   General, Core, and Other are also allowed as types

2. OS type and version:  Linux
3. Java version: 11.0.10 / Kotlin
4. logging version(s):  3.1.0 , 3.0.1

#### Steps to reproduce

  1.  Take code from github example on main readme page and run it on a logName value with more than 1 page, it fails pulling the 2nd page.

#### Code example

```java

Page<LogEntry> entries = logging.listLogEntries(
    EntryListOption.filter("logName=projects/" + options.getProjectId() + "/logs/test-log"));
Iterator<LogEntry> entryIterator = entries.iterateAll().iterator();
while (entryIterator.hasNext()) {
  System.out.println(entryIterator.next());
}
```

#### Stack trace
```
com.google.cloud.logging.LoggingException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: page_token doesn't match arguments from the request

	at com.google.cloud.logging.spi.v2.GrpcLoggingRpc$2.apply(GrpcLoggingRpc.java:203)
	at com.google.cloud.logging.spi.v2.GrpcLoggingRpc$2.apply(GrpcLoggingRpc.java:197)
	at com.google.api.core.ApiFutures$ApiFunctionToGuavaFunction.apply(ApiFutures.java:240)
	at com.google.common.util.concurrent.AbstractCatchingFuture$CatchingFuture.doFallback(AbstractCatchingFuture.java:223)
	at com.google.common.util.concurrent.AbstractCatchingFuture$CatchingFuture.doFallback(AbstractCatchingFuture.java:211)
	at com.google.common.util.concurrent.AbstractCatchingFuture.run(AbstractCatchingFuture.java:124)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1213)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:983)
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:771)
	at com.google.api.gax.retrying.BasicRetryingFuture.handleAttempt(BasicRetryingFuture.java:200)
	at com.google.api.gax.retrying.CallbackChainRetryingFuture$AttemptCompletionListener.handle(CallbackChainRetryingFuture.java:135)
	at com.google.api.gax.retrying.CallbackChainRetryingFuture$AttemptCompletionListener.run(CallbackChainRetryingFuture.java:117)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1213)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:983)
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:771)
	at com.google.api.core.AbstractApiFuture$InternalSettableFuture.setException(AbstractApiFuture.java:95)
	at com.google.api.core.AbstractApiFuture.setException(AbstractApiFuture.java:77)
	at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
	at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:68)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1074)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1213)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:983)
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:771)
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:563)
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:533)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:557)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:69)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:738)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:717)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: page_token doesn't match arguments from the request
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:49)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:72)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:60)
	... 18 more
Caused by: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: page_token doesn't match arguments from the request
	at io.grpc.Status.asRuntimeException(Status.java:535)
	... 10 more
```

#### External references such as API reference guides

- ?

#### Any additional information below


Following these steps guarantees the quickest resolution possible.

Thanks!

--- a/.readme-partials.yaml
+++ b/.readme-partials.yaml
@@ -94,7 +94,7 @@ custom_content: |
   imports at the top of your file:
   
   ```java
-  import com.google.cloud.Page;
+  import com.google.api.gax.paging.Page;
   import com.google.cloud.logging.LogEntry;
   import com.google.cloud.logging.Logging.EntryListOption;
   ```
@@ -102,14 +102,14 @@ custom_content: |
   Then, to list the log entries, use the following code:
 
   ``` java
-  Page<LogEntry> entries = logging.listLogEntries(
-      EntryListOption.filter("logName=projects/" + options.getProjectId() + "/logs/test-log"));
-  Iterator<LogEntry> entryIterator = entries.iterateAll().iterator();
-  while (entryIterator.hasNext()) {
-    System.out.println(entryIterator.next());
+  Page<LogEntry> entries = logging.listLogEntries(EntryListOption.filter(logFilter));
+  while (entries != null) {
+    for (LogEntry logEntry : entries.iterateAll()) {
+      System.out.println(logEntry);
+    }
+    entries = entries.getNextPage();
   }
   ```
-
   #### Add a Cloud Logging handler to a logger
   
   You can also register a `LoggingHandler` to a `java.util.logging.Logger` that publishes log entries
@@ -133,13 +133,3 @@ custom_content: |
   ```
   com.google.cloud.examples.logging.snippets.AddLoggingHandler.handlers=com.google.cloud.logging.LoggingHandler
   ```
-  
-  #### Complete source code
-
-  In
-  [CreateAndListMetrics.java](https://github.com/googleapis/google-cloud-java/tree/master/google-cloud-examples/src/main/java/com/google/cloud/examples/logging/snippets/CreateAndListMetrics.java),
-  [WriteAndListLogEntries.java](https://github.com/googleapis/google-cloud-java/tree/master/google-cloud-examples/src/main/java/com/google/cloud/examples/logging/snippets/WriteAndListLogEntries.java)
-  and
-  [AddLoggingHandler.java](https://github.com/googleapis/google-cloud-java/tree/master/google-cloud-examples/src/main/java/com/google/cloud/examples/logging/snippets/AddLoggingHandler.java)
-  we put together all the code shown above into three programs. The programs assume that you are
-  running on Compute Engine or from your own desktop.

--- a/.readme-partials.yaml
+++ b/.readme-partials.yaml
@@ -102,7 +102,8 @@ custom_content: |
   Then, to list the log entries, use the following code:
 
   ``` java
-  Page<LogEntry> entries = logging.listLogEntries(EntryListOption.filter(logFilter));
+  Page<LogEntry> entries = logging.listLogEntries(
+      EntryListOption.filter("logName=projects/" + options.getProjectId() + "/logs/test-log"));
   while (entries != null) {
     for (LogEntry logEntry : entries.iterateAll()) {
       System.out.println(logEntry);

--- a/README.md
+++ b/README.md
@@ -202,7 +202,8 @@ import com.google.cloud.logging.Logging.EntryListOption;
 Then, to list the log entries, use the following code:
 
 ``` java
-Page<LogEntry> entries = logging.listLogEntries(EntryListOption.filter(logFilter));
+Page<LogEntry> entries = logging.listLogEntries(
+    EntryListOption.filter("logName=projects/" + options.getProjectId() + "/logs/test-log"));
 while (entries != null) {
   for (LogEntry logEntry : entries.iterateAll()) {
     System.out.println(logEntry);

--- a/README.md
+++ b/README.md
@@ -194,7 +194,7 @@ With Logging you can also list log entries that have been previously written. Ad
 imports at the top of your file:
 
 ```java
-import com.google.cloud.Page;
+import com.google.api.gax.paging.Page;
 import com.google.cloud.logging.LogEntry;
 import com.google.cloud.logging.Logging.EntryListOption;
 ```
@@ -202,14 +202,14 @@ import com.google.cloud.logging.Logging.EntryListOption;
 Then, to list the log entries, use the following code:
 
 ``` java
-Page<LogEntry> entries = logging.listLogEntries(
-    EntryListOption.filter("logName=projects/" + options.getProjectId() + "/logs/test-log"));
-Iterator<LogEntry> entryIterator = entries.iterateAll().iterator();
-while (entryIterator.hasNext()) {
-  System.out.println(entryIterator.next());
+Page<LogEntry> entries = logging.listLogEntries(EntryListOption.filter(logFilter));
+while (entries != null) {
+  for (LogEntry logEntry : entries.iterateAll()) {
+    System.out.println(logEntry);
+  }
+  entries = entries.getNextPage();
 }
 ```
-
 #### Add a Cloud Logging handler to a logger
 
 You can also register a `LoggingHandler` to a `java.util.logging.Logger` that publishes log entries
@@ -234,15 +234,6 @@ file. Adding, for instance, the following line:
 com.google.cloud.examples.logging.snippets.AddLoggingHandler.handlers=com.google.cloud.logging.LoggingHandler
 ```
 
-#### Complete source code
-
-In
-[CreateAndListMetrics.java](https://github.com/googleapis/google-cloud-java/tree/master/google-cloud-examples/src/main/java/com/google/cloud/examples/logging/snippets/CreateAndListMetrics.java),
-[WriteAndListLogEntries.java](https://github.com/googleapis/google-cloud-java/tree/master/google-cloud-examples/src/main/java/com/google/cloud/examples/logging/snippets/WriteAndListLogEntries.java)
-and
-[AddLoggingHandler.java](https://github.com/googleapis/google-cloud-java/tree/master/google-cloud-examples/src/main/java/com/google/cloud/examples/logging/snippets/AddLoggingHandler.java)
-we put together all the code shown above into three programs. The programs assume that you are
-running on Compute Engine or from your own desktop.
 
 
 

